onnxruntime/core/framework/prepacked_weights_container.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/prepacked_weights_container.h"
+#include "core/framework/allocatormgr.h"
+
+namespace onnxruntime {
+
+class BufferHasher {
+ public:
+  std::size_t operator()(const int8_t* data, size_t num_bytes) const {
+    std::size_t ret = num_bytes;
+    for (size_t i = 0; i < num_bytes; ++i) {
+      ret ^= std::hash<int32_t>()(data[i]);
+    }
```

Comment:
I'd suggest using the MurmurHash class as per KernelDef::CalculateHash. 

Failing that, at a minimum I would have thought you'd need to so some bit shifting like this: [https://stackoverflow.com/a/2595226/684911](https://stackoverflow.com/a/2595226/684911)